server/server.go

```diff
@@ -2418,8 +2411,26 @@ func (s *StanServer) processCloseRequest(m *nats.Msg) {
 		return
 	}
 
+	// In clustered mode, drop the request if we're not the metadata leader.
+	// Another server will handle it.
+	if s.isClustered() && !s.isMetadataLeader() {
+		return
+	}
+
+	// If clustered, thread operations through Raft.
+	if s.isClustered() {
+		if err := s.replicateConnClose(req.ClientID, m.Reply, true); err != nil {
+			s.log.Errorf("Failed to replicate close request: %v", err)
+			s.sendCloseErr(m.Reply, err.Error())
+		}
+	} else {
+		s.scheduleConnClose(req.ClientID, m.Reply)
+	}
+}
+
+func (s *StanServer) scheduleConnClose(clientID, reply string) {
 	// Create the control and corresponding NATS message
-	ctrlMsg, ctrlNatsMsg := s.createCtrlMsg(spb.CtrlMsg_ConnClose, true, m.Reply, []byte(req.ClientID))
+	ctrlMsg, ctrlNatsMsg := s.createCtrlMsg(spb.CtrlMsg_ConnClose, true, reply, []byte(clientID))
```

Comment:
`reply` should not be set (or reset to empty) if the server processing this is not the leader.